src/bootstrap/src/core/config/config.rs

```diff
@@ -1008,6 +1008,15 @@ impl Config {
             | Subcommand::Vendor { .. } => flags.stage.unwrap_or(0),
         };
 
+        // Now check that the selected stage makes sense, and if not, print a warning and end
+        match (config.stage, &config.cmd) {
+            (0, Subcommand::Build) => {
+                eprintln!("WARNING: cannot build anything on stage 0. Use at least stage 1.");
+                exit!(1)
+            }
+            _ => {}
+        }
```

Comment:
Question: if we warn on the no-op that is `./x build --stage=0`, what about `./x check --stage=0`? I guess for the check case, we can't do that yet without untying bootstrap tool checking from staging?

Also, does this reject e.g. `./x build src/tools/opt-dist --stage=0`?

EDIT: indeed,

```rs
    #[test]
    #[should_panic]
    fn build_bootstrap_tool_stage_0() {
        insta::assert_snapshot!(get_steps("opt-dist", "build", Some(0)), @"");
    }
```

Reply:
Check is a whole another beast, I'm working on a PR for it, but the individual bootstrap kinds (build/check/etc.) are mostly separate in the stage handling, so I will do it piece by piece.

Reply:
Btw, I'm planning for `check --stage 0` to also be an error.